tests/functional/glusterd/test_gluster_operations_after_removing_firewall.py

```diff
@@ -0,0 +1,241 @@
+from glustolibs.gluster.gluster_base_class import GlusterBaseClass
+from glustolibs.gluster.volume_ops import (volume_create, volume_start,
+                                           get_volume_list)
+from glustolibs.gluster.volume_libs import (cleanup_volume)
+from glustolibs.gluster.peer_ops import (peer_probe, peer_detach,
+                                         peer_probe_servers,
+                                         nodes_from_pool_list,
+                                         wait_for_peers_to_connect,
+                                         is_peer_connected)
+from glustolibs.gluster.lib_utils import (form_bricks_list,
+                                          add_services_to_firewall,
+                                          remove_service_from_firewall)
+from glustolibs.gluster.exceptions import ExecutionError
+from glustolibs.gluster.mount_ops import mount_volume
+
+
+class TestGlusterOperationsAfterRemovingFirewall(GlusterBaseClass):
+
+    def setUp(self):
+
+        # Adding services list here, so that it can be
+        # used in teardown if setup fails
+        self.services_lst = ['glusterfs', 'nfs', 'rpc-bind']
+
+        # Performing peer detach
+        for server in self.servers[1:]:
+            ret, _, _ = peer_detach(self.mnode, server)
+            if ret != 0:
+                raise ExecutionError("Peer detach failed")
+        self.get_super_method(self, 'setUp')()
+
+    def tearDown(self):
+        # Reset firewall services to the zone
+        if not self.firewall_added:
+            ret = add_services_to_firewall(self.servers[:2],
+                                           self.services_lst, True)
+            if not ret:
+                raise ExecutionError("Failed to add firewall services")
+
+        # Reload firewall services
+        ret = self._reload_firewall_service(self.servers[:2])
+        if not ret:
+            raise ExecutionError("Failed to reload firewall services")
+
+        # Cleanup the volumes
+        vol_list = get_volume_list(self.mnode)
+        if vol_list is not None:
+            for volume in vol_list:
+                ret = cleanup_volume(self.mnode, volume)
+                if not ret:
+                    raise ExecutionError("Failed to cleanup volume")
+                g.log.info("Volume deleted successfully : %s", volume)
+
+        # Peer probe detached servers
+        pool = nodes_from_pool_list(self.mnode)
+        for node in pool:
+            peer_detach(self.mnode, node)
+        ret = peer_probe_servers(self.mnode, self.servers)
+        if not ret:
+            raise ExecutionError("Failed to probe detached "
+                                 "servers %s" % self.servers)
+        g.log.info("Peer probe success for detached "
+                   "servers %s", self.servers)
+        self.get_super_method(self, 'tearDown')()
+
+    def _reload_firewall_service(self, nodes):
+        # Reload the firewall service on the nodes
```

Comment:
Instead of a comment can we make it a docstring `""Reload the firewall service on the nodes"""`

Reply:
Ack